### PR TITLE
Edit profile pages post

### PR DESCRIPTION
**Pull request by CalamariDude**
I might need to double check indentation.

### CODE REVIEW

File: src/main/java/codeu/controller/UsersServlet.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  /**
+     * This function fires when a user requests the /users URL. It simply forwards
+     * the request to users.jsp.
+     */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+	String username = (String) request.getSession().getAttribute("user");
+	if (username == null) {
```
Comment (hsha0) [inline src/main/java/codeu/controller/UsersServlet.java]: It's in my part, lol. Indentation should be 4 spaces

File: src/main/java/codeu/controller/UsersServlet.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      // user is not logged in, don't let them go to profile page
+      response.sendRedirect("/login");
+      return;
+    }
+	
+	User user = userStore.getUser(username);
```
Comment (hsha0) [inline src/main/java/codeu/controller/UsersServlet.java]: My part again. Indentation should be 4 spaces

File: src/main/java/codeu/controller/UsersServlet.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    if (user == null) {
+      // user was not found, don't let them go to profile page
+      response.sendRedirect("/login");
+      return;
+    }
+    
+    request.setAttribute("user", user);
+	  request.getRequestDispatcher("/WEB-INF/view/users.jsp").forward(request, response);
```
Comment (hsha0) [inline src/main/java/codeu/controller/UsersServlet.java]: indentation should be 4 spaces. (I must mess up with space and tab before...)

File: src/main/java/codeu/controller/UsersServlet.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      request.getRequestDispatcher("/WEB-INF/view/about.jsp").forward(request, response);
+      response.sendRedirect("/login");
+      return;
+    }
+
+
+    User.Builder userBuilder = new User.Builder( user.getId(), user.getName(), user.getPassword(), user.getCreationTime());
+    userBuilder.setAge(age);
+    userBuilder.setFirstName(user.getFirstName());
+    userBuilder.setLastName(user.getLastName());
```
Comment (hsha0) [inline src/main/java/codeu/controller/UsersServlet.java]: I don't think we have the first name and the last name before. They should also be assigned through request.getParameter.

File: src/main/java/codeu/controller/UsersServlet.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    userBuilder.setPhoneNum(phoneNum);
+    userBuilder.setBio(bio);
+    userBuilder.setPictureURL(pictureURL);
+    user = userBuilder.createUser();
+
+    userStore.updateUser(user);
+
+    response.sendRedirect("/about");
+  }
+}
```
Comment (hsha0) [inline src/main/java/codeu/controller/UsersServlet.java]: I changed the filename to UserServlet.java. Be careful when you solve the conflicts.

File: src/main/java/codeu/model/data/User.java [hunk 6]
```diff
@@ -29,6 +29,7 @@
     private final String email;
     private final String phoneNum;
     private final String bio;
+		private final String pictureURL;
```
Comment (hsha0) [inline src/main/java/codeu/model/data/User.java]: Indentation here 6 spaces

File: src/main/java/codeu/model/data/User.java [hunk 7]
```diff
@@ -66,9 +68,10 @@ public User(UUID newId, String newName, String newPassword, Instant newCreation)
      * @param newEmail the email of this User
      * @param newPhoneNum the phone number of this User
      * @param newBio the bio of this User
+		 * @param newPictureURL the URL of the profile picture of this User
```
Comment (hsha0) [inline src/main/java/codeu/model/data/User.java]: indentation here 6 spaces

File: src/main/java/codeu/model/data/User.java [hunk 8]
```diff
@@ -206,4 +216,9 @@ public String getPhoneNum() {
     public String getBio() {
 	return bio;
     }
+
+		/** Returns the URL of the profile picture of this User. */
+		public String getPictureURL() {
+			return pictureURL;
+		}
 }
```
Comment (hsha0) [inline src/main/java/codeu/model/data/User.java]: All indentations should be 2-space standard

File: src/test/java/codeu/controller/UserServletTest.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        User user = new User(randomUUID,"newusername", "password", Instant.now());
+        Mockito.when(mockUserStore.getUser("newusername")).thenReturn(user);
+
+        usersServlet.doPost(mockRequest, mockResponse);
+
+        ArgumentCaptor<User> userArgumentCaptor = ArgumentCaptor.forClass(User.class);
+        Mockito.verify(mockUserStore).updateUser(userArgumentCaptor.capture());
+
+        Assert.assertEquals(userArgumentCaptor.getValue().getName(), "newusername");
+        Assert.assertEquals(userArgumentCaptor.getValue().getPictureURL(), "www.url.com");
```
Comment (hsha0) [inline src/test/java/codeu/controller/UserServletTest.java]: We also need to check the first name and the last name, right?

File: src/test/java/codeu/controller/UserServletTest.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        Assert.assertEquals(userArgumentCaptor.getValue().getBio(), "Example text about a bio.");
+        Assert.assertTrue(userArgumentCaptor.getValue().getAge() == 20);
+        Assert.assertEquals(userArgumentCaptor.getValue().getEmail(), "example@xyz.com");
+        Assert.assertEquals(userArgumentCaptor.getValue().getPhoneNum(), "3335551111");
+
+        Mockito.verify(mockResponse).sendRedirect("/about");
```
Comment (hsha0) [inline src/test/java/codeu/controller/UserServletTest.java]: I used /user in my UserServlet.java. You can choose either one of them when merging.

**Review by hsha0 — CHANGES_REQUESTED** [review]

File: src/test/java/codeu/controller/UserServletTest.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  @Test
+  public void testDoPostUpdatesUserCorrectly() throws IOException, ServletException {
+
+    Mockito.when(mockRequest.getParameter("user")).thenReturn("newusername");
+    Mockito.when(mockRequest.getParameter("pictureurl")).thenReturn("www.url.com");
+    Mockito.when(mockRequest.getParameter("bio")).thenReturn("Example text about a bio.");
+    Mockito.when(mockRequest.getParameter("age")).thenReturn("20");
+    Mockito.when(mockRequest.getParameter("email")).thenReturn("example@xyz.com");
+    Mockito.when(mockRequest.getParameter("phone")).thenReturn("3335551111");
+
```
Comment (hsha0) [inline src/test/java/codeu/controller/UserServletTest.java:58]: getParameter("first") and getParameter("last")

File: src/test/java/codeu/controller/UserServletTest.java [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    ArgumentCaptor<User> userArgumentCaptor = ArgumentCaptor.forClass(User.class);
+    Mockito.verify(mockUserStore).updateUser(userArgumentCaptor.capture());
+
+    Assert.assertEquals(userArgumentCaptor.getValue().getName(), "newusername");
+    Assert.assertEquals(userArgumentCaptor.getValue().getPictureURL(), "www.url.com");
+    Assert.assertEquals(userArgumentCaptor.getValue().getBio(), "Example text about a bio.");
+    Assert.assertTrue(userArgumentCaptor.getValue().getAge() == 20);
+    Assert.assertEquals(userArgumentCaptor.getValue().getEmail(), "example@xyz.com");
+    Assert.assertEquals(userArgumentCaptor.getValue().getPhoneNum(), "3335551111");
+
```
Comment (hsha0) [inline src/test/java/codeu/controller/UserServletTest.java]: assertEquals for first name and last name

**Review by hsha0 — COMMENTED** [review]
Need tests for first name and last name, other LGTM. You can merge after adding the tests.

**Review by hsha0 — APPROVED** [review]

**Comment by hsha0** [comment]
Also need to add Assert.assertEquals() for first and last name

**Comment by hsha0** [comment]
I am wondering why when Running codeu.model.store.persistence.PersistentDataStoreTest, it shows INFO: Local Datastore initialized: 
	Type: High Replication
	Storage: In-memory

**Comment by CalamariDude** [comment]
Hmmm, looked around for significant places where there could be high usage, couldn't find anything. Tried googling it, seems like its a datastore-handler related problem.  @TylerSandman any ideas?

**Comment by hsha0** [comment]
And for adding the user to persistenceDataStore, why we don't use addUser() function?
/** Add a new user to the current set of users known to the application. */
  public void addUser(User user) {
    users.add(user);
    persistentStorageAgent.writeThrough(user);
  }

**Comment by CalamariDude** [comment]
You mean for updating the user?

**Comment by hsha0** [comment]
Yeah, can we just use addUser function instead of updateUser?

**Pull request merged**